concourse/scripts/tag_release_test.go

```diff
@@ -0,0 +1,219 @@
+package scripts_test
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("TagRelease", func() {
+
+	var (
+		workingDirectory string
+		originRepoPath   string
+		baseCmd          *exec.Cmd
+	)
+
+	BeforeEach(func() {
+		gitVersionCommand := exec.Command("git", "--version")
+
+		session, err := gexec.Start(gitVersionCommand, GinkgoWriter, GinkgoWriter)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(session).Should(gexec.Exit(0))
+		Expect(session.Out.Contents()).To(MatchRegexp(".* [2-9]\\.[0-9]+\\.[0-9]+.*"), "Expected git version 2.0.0 or bigger")
+		Expect(session.Err.Contents()).To(BeEmpty())
+	})
+
+	BeforeEach(func() {
+		workingDirectory, baseCmd = prepareWorkingDirectory(
+			"./tag_release.sh",
+		)
+
+		prepareSshKeys(baseCmd)
+		configGitEmailAndName(baseCmd)
+
+		originRepoPath = setupClonedRepository(baseCmd)
+		baseCmd.Env = append(baseCmd.Env, fmt.Sprintf("GIT_REPO_URL=%s", originRepoPath))
+	})
+
+	AfterEach(func() {
+		cleanUpWorkingDirectory(workingDirectory)
+	})
+
+	Context("when there are multiple 'previous-*' tags to be promoted", func() {
+		BeforeEach(func() {
+			runBashScript(baseCmd, `
+        cd origin_repo
```

Comment:
Yeah, it is strange, in theory I am using spaces only and my vim dsplays it OK. I will check.